### PR TITLE
Draft JEP: Convert modules to plugins

### PR DESCRIPTION
**Pull request by jglick**
Provides background and refreshed proposals for https://github.com/jenkinsci/jenkins/pull/3988.

### CODE REVIEW

File: jep/0000/README.adoc [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+If this decision needs to be reversed in the future,
+`InstanceIdentityRSAProvider` or its equivalent could just be added to core
+and `instance-identity` made an empty deprecated “tombstone”.
+
+=== Keeping `ssh-cli-auth` separate
+
+As of the removal of Remoting-based CLI,
+`ssh-cli-auth` merely provided utility classes used by `sshd`,
+with no independent functionality.
+Therefore there is no advantage in offering it as a separate plugin.
```
Comment (kuisathaverat) [inline jep/0000/README.adoc:161]: Do you purpose to join ssh-cli-auth-module to ssh-module?

Comment (jglick) [reply jep/0000/README.adoc:161]: Yes, as in **Specification**.

**Comment by oleg-nenashev** [comment]
Oops. Could you please update it to JEP-230?

**Comment by jglick** [comment]
@oleg-nenashev are you asking something of me? Not sure I understand.

JEP-1 is vague but my advice for accepting JEPs as draft would be to assign a number as part of the merge. So for example:

```bash
# in checkout of jep@master
git pull --no-commit --no-ff https://github.com/$sponsor/jep $branch
perl -pie s/0000/$num/g jep/0000/README.adoc
git mv jep/0000 jep/$num
bash jep/set-jep-status $num Draft
git commit -a "Accepting JEP-$num as draft"
git push
```

**Comment by jglick** [comment]
@jenkinsci/jep-editors ping

**Review by omehegan — APPROVED** [review]
The proposal is clear and sounds reasonable to me.

**Comment by jglick** [comment]
Can it be merged as a draft then? Is there something holding this up?

@oleg-nenashev FYI I had to force-push to restore proper contents; something weird happened with the attempted merge. Recommend standardizing a script like https://github.com/jenkinsci/jep/pull/326#issuecomment-755733671.

**Comment by omehegan** [comment]
I see no reason to hold this up, but I have not been very involved in the JEP process lately, so I hesitate to make any authoritative statements about things :)

**Comment by jglick** [comment]
Then @bitwiseman perhaps?

**Review by oleg-nenashev — APPROVED** [review]
LGTM as a JEP draft. Will integrate it tonight

**Pull request merged**